v-next/hardhat-network-helpers/src/internal/network-helpers/network-helpers.ts

```diff
@@ -29,93 +33,162 @@ import { Time } from "./time/time.js";
+    await this.throwIfNotDevelopmentNetwork();
     return mineUpTo(this.#provider, blockNumber, this.time);
   }
 
-  public reset(url?: string, blockNumber?: NumberLike): Promise<void> {
+  public async reset(url?: string, blockNumber?: NumberLike): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return reset(this, this.#provider, url, blockNumber);
   }
 
-  public setBalance(address: string, balance: NumberLike): Promise<void> {
+  public async setBalance(address: string, balance: NumberLike): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setBalance(this.#provider, address, balance);
   }
 
-  public setBlockGasLimit(blockGasLimit: NumberLike): Promise<void> {
+  public async setBlockGasLimit(blockGasLimit: NumberLike): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setBlockGasLimit(this.#provider, blockGasLimit);
   }
 
-  public setCode(address: string, code: string): Promise<void> {
+  public async setCode(address: string, code: string): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setCode(this.#provider, address, code);
   }
 
-  public setCoinbase(address: string): Promise<void> {
+  public async setCoinbase(address: string): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setCoinbase(this.#provider, address);
   }
 
-  public setNextBlockBaseFeePerGas(baseFeePerGas: NumberLike): Promise<void> {
+  public async setNextBlockBaseFeePerGas(
+    baseFeePerGas: NumberLike,
+  ): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setNextBlockBaseFeePerGas(this.#provider, baseFeePerGas);
   }
 
-  public setNonce(address: string, nonce: NumberLike): Promise<void> {
+  public async setNonce(address: string, nonce: NumberLike): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setNonce(this.#provider, address, nonce);
   }
 
-  public setPrevRandao(prevRandao: NumberLike): Promise<void> {
+  public async setPrevRandao(prevRandao: NumberLike): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setPrevRandao(this.#provider, prevRandao);
   }
 
-  public setStorageAt(
+  public async setStorageAt(
     address: string,
     index: NumberLike,
     value: NumberLike,
   ): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return setStorageAt(this.#provider, address, index, value);
   }
 
-  public stopImpersonatingAccount(address: string): Promise<void> {
+  public async stopImpersonatingAccount(address: string): Promise<void> {
+    await this.throwIfNotDevelopmentNetwork();
     return stopImpersonatingAccount(this.#provider, address);
   }
 
-  public takeSnapshot(): Promise<SnapshotRestorer> {
+  public async takeSnapshot(): Promise<SnapshotRestorer> {
+    await this.throwIfNotDevelopmentNetwork();
     return takeSnapshot(this.#provider);
   }
+
+  public async throwIfNotDevelopmentNetwork(): Promise<void> {
```

Comment:
The method is marked as public because it also needs to be used in the `Time` class. However, it will not appear when a user lists available methods during coding, as it is not included in the `NetworkHelpersI` interface.

Other solutions considered:
- Marking the method as private and passing it through the constructor of the `Time` class.
- Applying a decorator to the class that would automatically enforce the check logic on every method.

Reply:
Got it! Makes sense, thanks for giving me the extra context. In that case, I think the current implementation totally works.